### ISSUE TITLE
Fix FP S2819 (`cross-origin checks`): Validation missed when in another function

### ISSUE THREAD

**Issue by gaetan-ferry-sonarsource** [SonarSource/SonarJS#3788]
# WHAT
S2819 detects cross-origin message handlers that do not validate an event's origin. When the validation is performed directly in the event handler's code, no issue is raised, which matches the specification.
However, when the validation is performed in another function called from the handler, the analyzer misses it and raise an issue.

# Example code
```javascript
const Class = function () {
  
  Class.prototype.falsePositive = function () {
    window.addEventListener("message", event => { // FP
      if (!this.validateEvent(event)) {
        return;
      }

      document.dispatchEvent(event);
    });
  };

  Class.prototype.trueNegative = function () {
    window.addEventListener("message", event => { // TN
      if (!event.origin == "https://example.org/") {
        return;
      }

      document.dispatchEvent(event);
    });
  };

  Class.prototype.validateEvent = function (event) {
    return (event.origin == "https://example.org/")
  };
}
```
# Note
This false positive has been highlighted by a customer on SonarCloud

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-98
